spring-cloud-starter-stream-source-sftp-dataflow/src/main/java/org/springframework/cloud/stream/app/sftp/dataflow/source/SftpDataflowSourceConfiguration.java

```diff
@@ -0,0 +1,270 @@
+
+		IntegrationFlowBuilder flowBuilder;
+
+		if (properties.isListOnly()) {
+			return listingFlow(sftpSessionFactory);
+		}
+
+		else {
+			//Save remote file to local file system
+			SftpInboundChannelAdapterSpec messageSourceBuilder = Sftp.inboundAdapter(
+				this.properties.isMultiSource() ? this.delegatingSessionFactory.getFactory() : sftpSessionFactory)
+				.preserveTimestamp(this.properties.isPreserveTimestamp())
+				.remoteDirectory(this.properties.getRemoteDir())
+				.remoteFileSeparator(this.properties.getRemoteFileSeparator())
+				.localDirectory(new File(properties.getLocalDir().getPath()))
+				.autoCreateLocalDirectory(this.properties.isAutoCreateLocalDir())
+				.temporaryFileSuffix(this.properties.getTmpFileSuffix())
+				.deleteRemoteFiles(this.properties.isDeleteRemoteFiles())
+				.filter(filterChain);
+			if (this.properties.getMaxFetch() != null) {
+				messageSourceBuilder.maxFetchSize(this.properties.getMaxFetch());
+			}
+
+			flowBuilder = IntegrationFlows.from(messageSourceBuilder, consumerSpec(this.fileSourceRotator));
+
+		}
+
+		flowBuilder.channel(taskLaunchRequestChannel());
+
+		return flowBuilder.get();
+	}
+
+	private IntegrationFlow listingFlow(SessionFactory<LsEntry> sftpSessionFactory) {
+		IntegrationFlowBuilder builder = this.properties.isMultiSource() ?
+			multiSourceListingFlowBuilder() :
+			singleSourceListingFlowBuilder(sftpSessionFactory);
+		return builder.get();
+	}
+
+	private IntegrationFlowBuilder singleSourceListingFlowBuilder(SessionFactory<LsEntry> sftpSessionFactory) {
+		return IntegrationFlows.from(() -> this.properties.getRemoteDir(), consumerSpec(this
+			.listFilesRotator))
+			.handle(Sftp.outboundGateway(sftpSessionFactory, AbstractRemoteFileOutboundGateway.Command.LS.getCommand(),
+				"payload").options(AbstractRemoteFileOutboundGateway.Option.NAME_ONLY.getOption()))
+			.split()
+			.channel(sftpListInputChannel());
+
+	}
+
+	private IntegrationFlowBuilder multiSourceListingFlowBuilder() {
+		IntegrationFlowBuilder flow = IntegrationFlows.from(() -> this.listFilesRotator.getCurrentDirectory(),
+			consumerSpec(this.listFilesRotator))
+			.handle(Sftp.outboundGateway(this.delegatingSessionFactory.getFactory(),
+				AbstractRemoteFileOutboundGateway.Command.LS.getCommand(), "payload")
+				.options(AbstractRemoteFileOutboundGateway.Option.NAME_ONLY.getOption()));
+
+		return flow.handle(this.listFilesRotator, "clearKey")
+			.split()
+			.channel(sftpListInputChannel());
+	}
+
+	private Consumer<SourcePollingChannelAdapterSpec> consumerSpec(Advice advice) {
+		if (advice == null) {
+			return spec -> spec.poller(this.defaultPoller);
+		}
+		else {
+			PollerMetadata poller = new PollerMetadata();
+			BeanUtils.copyProperties(this.defaultPoller, poller, "transactionSynchronizationFactory");
+			poller.setAdviceChain(Arrays.asList(advice));
+			return spec -> spec.poller(poller);
+		}
+	}
+
+	private Consumer<SourcePollingChannelAdapterSpec> consumerSpecWithDelete(Advice advice) {
```

Comment:
This method is out of use. Looks like a `deleteRemoteFiles` configuration property is not honored.